### PR TITLE
Implement solomachine chain context

### PR DESCRIPTION
**Pull request by soareschen**
Closes:  informalsystems/hermes-sdk#30

## Description

This PR implements an MVP solomachine chain context, which implements `OfaIbcChain` for cross-chain communication between a solomachine chain and a Cosmos chain.

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-solomachine/src/impls/chain.rs [hunk 1]
```diff
@@ -0,0 +1,1365 @@
...
+            client_id,
+            *height,
+            &cosmos_consensus_state,
+        )
+        .map_err(Chain::encode_error)?;
+
+        let secret_key = self.chain.secret_key();
+
+        let connection_proof =
+            sign_with_data(secret_key, &client_state_data).map_err(Chain::encode_error)?;
```
Comment (ljoss17) [inline crates/relayer-solomachine/src/impls/chain.rs:488]:
```suggestion
            sign_with_data(secret_key, &connection_data).map_err(Chain::encode_error)?;
```
Shouldn't this be the proof for the connection_data?

Comment (soareschen) [reply crates/relayer-solomachine/src/impls/chain.rs:488]: Probably.

Comment (seanchen1991) [reply crates/relayer-solomachine/src/impls/chain.rs]: [7737304](https://github.com/informalsystems/hermes/pull/3509/commits/7737304fc38b281d5548de134c5b8e525400a928)

File: crates/relayer-solomachine/src/impls/chain.rs [hunk 2]
```diff
@@ -396,7 +395,44 @@ where
...
+            timestamp: self.chain.current_time(),
+            diversifier: new_diversifier,
+            data: packet_commitment_data_bytes,
+            path: path.as_bytes().to_vec(),
+        };
+
+        let proof = sign_with_data(secret_key, &sign_data).map_err(Chain::encode_error)?;
+
+        let payload = SolomachineTimeoutUnorderedPacketPayload {
+            update_height: *height,
```
Comment (seanchen1991) [inline crates/relayer-solomachine/src/impls/chain.rs:421]: I'm not sure if it's correct to give the `height` parameter as the `update_height` here. Looking at how the `CosmosReceivePacketPayload` and `CosmosTimeoutUnorderedPacketPayload` types are populated, their `update_heights` are taken from the packet proofs.

**Comment by soareschen** [comment]
This PR has been moved to https://github.com/informalsystems/hermes-sdk/pull/39.

**Pull request closed without merging**